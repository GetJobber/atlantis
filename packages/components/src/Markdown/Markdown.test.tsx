import React from "react";
import renderer from "react-test-renderer";
import { Markdown } from ".";

it("renders Text Component", () => {
  const tree = renderer.create(<Markdown content="Paragraph" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Italicized Component", () => {
  const tree = renderer
    .create(<Markdown content="_Italicized Foo_" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Bold Component", () => {
  const tree = renderer.create(<Markdown content="**Bold Foo**" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Level 1 Heading Component", () => {
  const tree = renderer.create(<Markdown content="# Heading 1" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Level 2 Heading Component", () => {
  const tree = renderer.create(<Markdown content="## Heading 2" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Level 3 Heading Component", () => {
  const tree = renderer.create(<Markdown content="### Heading 3" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Level 4 Heading Component", () => {
  const tree = renderer.create(<Markdown content="#### Heading 4" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Level 5 Heading Component", () => {
  const tree = renderer.create(<Markdown content="##### Heading 5" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Level 6 Heading Component", () => {
  const tree = renderer.create(<Markdown content="###### Heading " />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders a combination of Markdown components", () => {
  const content = `# This is a paragraph.

  The paragraph has some **bold text** and some _italicized text_.

  This is the end of this paragraph.`;
  const tree = renderer.create(<Markdown content={content} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders a without headings, paragraph, and wrapping content", () => {
  const content = `# No heading should show up

  This should just be pure string with **bold text** and some _italicized text_
  and of course a [link](http://to.somewhere)`;
  const tree = renderer
    .create(<Markdown content={content} basicUsage={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
