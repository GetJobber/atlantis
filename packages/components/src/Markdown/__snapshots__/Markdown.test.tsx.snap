// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Bold Component 1`] = `
<div
  className="padded base"
>
  <p
    className="base regular base text"
  >
    <b
      className="base bold"
    >
      Bold Foo
    </b>
  </p>
</div>
`;

exports[`renders Italicized Component 1`] = `
<div
  className="padded base"
>
  <p
    className="base regular base text"
  >
    <em
      className="base regular italic"
    >
      Italicized Foo
    </em>
  </p>
</div>
`;

exports[`renders Level 1 Heading Component 1`] = `
<div
  className="padded base"
>
  <h1
    className="base black jumbo uppercase heading"
  >
    Heading 1
  </h1>
</div>
`;

exports[`renders Level 2 Heading Component 1`] = `
<div
  className="padded base"
>
  <h2
    className="base black largest uppercase heading"
  >
    Heading 2
  </h2>
</div>
`;

exports[`renders Level 3 Heading Component 1`] = `
<div
  className="padded base"
>
  <h3
    className="base bold larger heading"
  >
    Heading 3
  </h3>
</div>
`;

exports[`renders Level 4 Heading Component 1`] = `
<div
  className="padded base"
>
  <h4
    className="base bold large heading"
  >
    Heading 4
  </h4>
</div>
`;

exports[`renders Level 5 Heading Component 1`] = `
<div
  className="padded base"
>
  <h5
    className="base bold base heading"
  >
    Heading 5
  </h5>
</div>
`;

exports[`renders Level 6 Heading Component 1`] = `
<div
  className="padded base"
>
  <h6>
    Heading
  </h6>
</div>
`;

exports[`renders Text Component 1`] = `
<div
  className="padded base"
>
  <p
    className="base regular base text"
  >
    Paragraph
  </p>
</div>
`;

exports[`renders a combination of Markdown components 1`] = `
<div
  className="padded base"
>
  <h1
    className="base black jumbo uppercase heading"
  >
    This is a paragraph.
  </h1>
  <p
    className="base regular base text"
  >
      The paragraph has some 
    <b
      className="base bold"
    >
      bold text
    </b>
     and some 
    <em
      className="base regular italic"
    >
      italicized text
    </em>
    .
  </p>
  <p
    className="base regular base text"
  >
      This is the end of this paragraph.
  </p>
</div>
`;

exports[`renders a without headings, paragraph, and wrapping content 1`] = `
Array [
  "No heading should show up",
  "  This should just be pure string with ",
  <b
    className="base bold"
  >
    bold text
  </b>,
  " and some ",
  <em
    className="base regular italic"
  >
    italicized text
  </em>,
  "
  and of course a ",
  <a
    href="http://to.somewhere"
  >
    link
  </a>,
]
`;
